pkg/controllers/management/node/controller.go

```diff
@@ -282,6 +282,20 @@ outer:
 }
 
 func (m *Lifecycle) Updated(obj *v3.Node) (*v3.Node, error) {
+	if isCustom(obj) {
+		m.setupCustom(obj)
+		newObj, err := v3.NodeConditionUpdated.Once(obj, func() (runtime.Object, error) {
+			if err := validateCustomHost(obj); err != nil {
+				return obj, err
+			}
+			m.setWaiting(obj)
```

Comment:
consider passing the message to setWaiting; the current message is hardcoded to `waiting to register with Kubernetes`, which is not quite true for this case.